## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-erasure.sh | :x: [more...](#28be3e6c-9ba9-4752-981b-2ce75b6baeb0) |
| mint-pools.sh | :x: [more...](#e6c8aa4a-98fd-4190-8b1d-73706b72c1ad) |
<a name="e6c8aa4a-98fd-4190-8b1d-73706b72c1ad"></a>
### 16795-1c6b645/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.1:32630
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 19bd164ba2e0:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 8 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 9 seconds
(5/14) Running awscli tests ... done in 1 minutes and 15 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... FAILED in 12 seconds
{
  "name": "mc",
  "duration": "11150",
  "function": "test_rb",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-30839 --quiet --no-color mb myminio/mc-test-bucket-10784-2  >>> \nmc: <ERROR> Unable to make bucket `myminio/mc-test-bucket-10784-2`. Put \"http://15.15.15.1:32630/mc-test-bucket-10784-2/\": dial tcp 15.15.15.1:32630: i/o timeout"
}
(7/14) Running minio-go tests ... done in 1 minutes and 11 seconds
(8/14) Running minio-java tests ... done in 37 seconds
(9/14) Running minio-js tests ... done in 52 seconds
(10/14) Running minio-py tests ... FAILED in 3 minutes and 6 seconds
{
  "name": "minio-py:test_presigned_get_object_default_expiry",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-1b181e88-4a95-4cb3-82f9-2ec019b30d89",
    "object_name": "ad14f40c-51ff-4631-9a4f-344aa094911c"
  },
  "message": "S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /minio-py-test-1b181e88-4a95-4cb3-82f9-2ec019b30d89, request_id: 174B750B1F98A053, host_id: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, bucket_name: minio-py-test-1b181e88-4a95-4cb3-82f9-2ec019b30d89",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1269, in test_presigned_get_object_default_expiry\n    _CLIENT.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 386, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketAlreadyOwnedByYou, message: Your previous request to create the named bucket succeeded and you already own it., resource: /minio-py-test-1b181e88-4a95-4cb3-82f9-2ec019b30d89, request_id: 174B750B1F98A053, host_id: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, bucket_name: minio-py-test-1b181e88-4a95-4cb3-82f9-2ec019b30d89\n",
  "duration": 56972
}
(10/14) Running s3cmd tests ... done in 17 seconds
(11/14) Running s3select tests ... done in 5 seconds
(12/14) Running versioning tests ... done in 3 minutes and 6 seconds

Executed 12 out of 14 tests successfully.
```
<a name="28be3e6c-9ba9-4752-981b-2ce75b6baeb0"></a>
### 16795-1c6b645/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.8:31271
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp c530c85025b3:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 7 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 8 seconds
(5/14) Running awscli tests ... done in 1 minutes and 14 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... FAILED in 17 seconds
{
  "name": "mc",
  "duration": "12179",
  "function": "test_mv_object",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-12670 --quiet --no-color mv myminio/mc-test-bucket-2927/dir-16081-4512/object-2 /tmp/tmp.IL6tHb6uBF/file  >>> \nmc: <ERROR> Unable to validate source `myminio/mc-test-bucket-2927/dir-16081-4512/object-2`."
}
(7/14) Running minio-go tests ... done in 49 seconds
(8/14) Running minio-java tests ... done in 18 seconds
(9/14) Running minio-js tests ... done in 47 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 28 seconds
(11/14) Running s3cmd tests ... done in 15 seconds
(12/14) Running s3select tests ... done in 2 seconds
(13/14) Running versioning tests ... done in 3 minutes and 15 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/metacache-set.go
+++ b/cmd/metacache-set.go
@@ -397,6 +397,11 @@ func (er *erasureObjects) streamMetadataParts(ctx context.Context, o listPathOpt
 	retries := 0
 	rpc := globalNotificationSys.restClientFromHash(pathJoin(o.Bucket, o.Prefix))
 
+	const (
+		retryDelay    = 50 * time.Millisecond
+		retryDelay250 = 250 * time.Millisecond
+	)
+
 	for {
 		if contextCanceled(ctx) {
 			return entries, ctx.Err()
@@ -411,7 +416,6 @@ func (er *erasureObjects) streamMetadataParts(ctx context.Context, o listPathOpt
 			retries = 1
 		}
 
-		const retryDelay = 250 * time.Millisecond
 		// All operations are performed without locks, so we must be careful and allow for failures.
 		// Read metadata associated with the object from a disk.
 		if retries > 0 {
@@ -425,7 +429,7 @@ func (er *erasureObjects) streamMetadataParts(ctx context.Context, o listPathOpt
 				_, err := disk.ReadVersion(ctx, minioMetaBucket,
 					o.objectPath(0), "", false)
 				if err != nil {
-					time.Sleep(retryDelay)
+					time.Sleep(retryDelay250)
 					retries++
 					continue
 				}
@@ -440,11 +444,19 @@ func (er *erasureObjects) streamMetadataParts(ctx context.Context, o listPathOpt
 			switch toObjectErr(err, minioMetaBucket, o.objectPath(0)).(type) {
 			case ObjectNotFound:
 				retries++
-				time.Sleep(retryDelay)
+				if retries == 1 {
+					time.Sleep(retryDelay)
+				} else {
+					time.Sleep(retryDelay250)
+				}
 				continue
 			case InsufficientReadQuorum:
 				retries++
-				time.Sleep(retryDelay)
+				if retries == 1 {
+					time.Sleep(retryDelay)
+				} else {
+					time.Sleep(retryDelay250)
+				}
 				continue
 			default:
 				return entries, fmt.Errorf("reading first part metadata: %w", err)
@@ -463,7 +475,7 @@ func (er *erasureObjects) streamMetadataParts(ctx context.Context, o listPathOpt
 				retries = -1
 			}
 			retries++
-			time.Sleep(retryDelay)
+			time.Sleep(retryDelay250)
 			continue
 		case errors.Is(err, io.EOF):
 			return entries, io.EOF
@@ -497,7 +509,7 @@ func (er *erasureObjects) streamMetadataParts(ctx context.Context, o listPathOpt
 						_, err := disk.ReadVersion(ctx, minioMetaBucket,
 							o.objectPath(partN), "", false)
 						if err != nil {
-							time.Sleep(retryDelay)
+							time.Sleep(retryDelay250)
 							retries++
 							continue
 						}
@@ -508,7 +520,7 @@ func (er *erasureObjects) streamMetadataParts(ctx context.Context, o listPathOpt
 				// Load partN metadata...
 				fi, metaArr, onlineDisks, err = er.getObjectFileInfo(ctx, minioMetaBucket, o.objectPath(partN), ObjectOptions{}, true)
 				if err != nil {
-					time.Sleep(retryDelay)
+					time.Sleep(retryDelay250)
 					retries++
 					continue
 				}
@@ -530,9 +542,9 @@ func (er *erasureObjects) streamMetadataParts(ctx context.Context, o listPathOpt
 			}()
 
 			tmp := newMetacacheReader(pr)
-			defer tmp.Close()
 			e, err := tmp.filter(o)
 			pr.CloseWithError(err)
+			tmp.Close()
 			entries.o = append(entries.o, e.o...)
 			if o.Limit > 0 && entries.len() > o.Limit {
 				entries.truncate(o.Limit)
@@ -546,11 +558,11 @@ func (er *erasureObjects) streamMetadataParts(ctx context.Context, o listPathOpt
 				switch toObjectErr(err, minioMetaBucket, o.objectPath(partN)).(type) {
 				case ObjectNotFound:
 					retries++
-					time.Sleep(retryDelay)
+					time.Sleep(retryDelay250)
 					continue
 				case InsufficientReadQuorum:
 					retries++
-					time.Sleep(retryDelay)
+					time.Sleep(retryDelay250)
 					continue
 				default:
 					logger.LogIf(ctx, err)

